### PR TITLE
[Merged by Bors] - chore: re-port Algebra.CharP.Basic

### PR DESCRIPTION
**Pull request by joneugster**
This forward-ports changes from leanprover-community/mathlib#11364

One unrelated downstream file times out, presumably due to the import graph subtly changing.

---

* [`algebra.char_p.basic`@`05a78c9451101108e638a0f213fb1bed82483545`..`ceb887ddf3344dab425292e497fa2af91498437c`](https://leanprover-community.github.io/mathlib-port-status/file/algebra/char_p/basic?range=05a78c9451101108e638a0f213fb1bed82483545..ceb887ddf3344dab425292e497fa2af91498437c)

This is the last file from [mathlib#11364](https://github.com/leanprover-community/mathlib/pull/11364). The other files are:
* [data.enat.basic](https://leanprover-community.github.io/mathlib-port-status/file/data/enat/basic) (synced in #3178 )
* [data.nat.multiplicity](https://leanprover-community.github.io/mathlib-port-status/file/data/nat/multiplicity) (ported after change)
* [ring_theory.multiplicity](https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/multiplicity) (ported after change)
* [ring_theory.polynomial.eisenstein.basic](https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/polynomial/eisenstein/basic) (ported after change)
* [combinatorics.additive.behrend](https://leanprover-community.github.io/mathlib-port-status/file/combinatorics/additive/behrend) (unported)

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/CharP/Basic.lean [hunk 1]
```diff
@@ -162,39 +233,23 @@ theorem eq_zero [CharZero R] : ringChar R = 0 :=
...
-      exact Nat.Prime.dvd_choose_self Fact.out h2 (Finset.mem_range.1 h1))
-    (by
-      intro h1
-      contrapose! h1
-      rw [Finset.mem_range]
-      exact Nat.Prime.pos Fact.out)
-  simp only [mul_one, one_mul, Nat.choose_zero_right, tsub_zero, Nat.cast_one, pow_zero]
+theorem add_pow_char_of_commute [Semiring R] {p : ℕ} [hp : Fact p.Prime] [CharP R p] (x y : R)
+    (h : Commute x y) : (x + y) ^ p = x ^ p + y ^ p :=
+  by
```
Comment (eric-wieser) [inline Mathlib/Algebra/CharP/Basic.lean:244]:
```suggestion
    (h : Commute x y) : (x + y) ^ p = x ^ p + y ^ p := by
```

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Comment by joneugster** [comment]
Thx Jeremy for fixing the newline + `by`. I'll leave the porting notes in for now. They don't hurt anybody and can always be dropped later.

bors r+

**Comment by eric-wieser** [comment]
bors merge

**Comment by eric-wieser** [comment]
Can you merge master?

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**